GitCommands/Submodules/SubmoduleInfoResult.cs

```diff
@@ -10,7 +10,7 @@ namespace GitCommands.Submodules
     public class SubmoduleInfoResult
     {
         // Module that was used to gather submodule info
-        public IGitModule Module { get; internal set; }
+        public GitModule Module { get; internal set; }
```

Comment:
Why?

Reply:
GetCurrentSubmoduleLocalPath() and SuperprojectModule() is not in IGitModule

Reply:
Passing currentModule around in addition to result (similar to how it was before)

Reply:
Please add them to the interface

Reply:
GetCurrentSubmoduleLocalPath() can be added, but SuperprojectModule() is not trivial (you have discussed this with @amaiorano related to the sidepanel) as the return type need to be changed which will cause an avalanche of other additions to IGitModule. So that should be a separate refactoring.

Reply:
A proposal in https://github.com/gerhardol/gitextensions/tree/feature/i6359-submodule-selective-update-interfaces
However, requires the use of casts...